perfkitbenchmarker/publisher.py

```diff
@@ -640,6 +650,90 @@ def _deDotKeys(self, res):
+    self.influx_uri = influx_uri
+    self.influx_db_name = influx_db_name
+
+  def PublishSamples(self, samples):
+    formated_samples = []
+    for sample in samples:
+      formated_samples.append(self._ConstructSample(sample))
+    self._Publish(formated_samples)
+
+  def _Publish(self, formated_samples):
+    try:
+      self._CreateDB()
+      body = ' \n '.join(formated_samples)
+      self._writeData(body)
+    except (IOError, httplib.HTTPException) as http_exception:
+      logging.debug("Error connecting to the database: ", http_exception)
+
+  def _ConstructSample(self, sample):
+    timestamp = str(int((10 ** 9) * sample['timestamp']))
+    measurement = 'perfkitbenchmarker'
+
+    tag_set_metadata = ''
+    if 'metadata' in sample:
+      if sample['metadata']:
+        tag_set_metadata = ','.join(self._FormatToKeyValue(sample['metadata']))
+    tag_keys = ('test', 'official', 'owner', 'run_uri', 'sample_uri',
+                'metric', 'unit')
+    tag_set = ','.join(self._FormatToKeyValue({k: sample[k] for k in tag_keys}))
+    if tag_set_metadata:
+      tag_set += ',' + tag_set_metadata
+
+    field_set = '%s=%s' % ('value', sample['value'])
+
+    sample_constructed_body = '%s,%s %s %s' % (measurement, tag_set,
+                                               field_set, timestamp)
+    return sample_constructed_body
+
+  def _FormatToKeyValue(self, sample):
+    key_value_pairs = []
+    for k, v in sample.iteritems():
+      key_value_pairs.append('%s=%s' % (k, v))
+    return key_value_pairs
+
+  def _CreateDB(self):
+    successful_http_request_codes = [200, 202, 204]
+    header = {'Content-type': 'application/x-www-form-urlencoded',
+              'Accept': 'text/plain'}
+    params = urllib.urlencode({'q': 'CREATE DATABASE ' + self.influx_db_name})
+    conn = httplib.HTTPConnection(self.influx_uri)
+    conn.request('POST', '/query?' + params, headers=header)
+    response = conn.getresponse()
+    response_status = response.status
+    response_response = response.response
+    if response_status in successful_http_request_codes:
+      logging.debug('Success!', self.influx_db_name, ' DB Created')
+    else:
+      logging.debug(response_status,
+                    'Request could not be completed due to: ',
+                    response_response)
+      raise httplib.HTTPException
+
+  def _WriteData(self, data):
+    successful_http_request_codes = [200, 202, 204]
+    params = data
+    header = {"Content-type": "application/octet-stream"}
+    conn = httplib.HTTPConnection(self.influx_uri)
+    conn.request('POST', '/write?' + 'db=' + self.influx_db_name, params,
+                 headers=header)
+    response = conn.getresponse()
+    response_status = response.status
+    response_response = response.response
+    if response_status in successful_http_request_codes:
+      logging.debug('writing samples to publisher: writing samples.')
+    else:
+      logging.debug(response_status,
```

Comment:
nitpick: Instead of using local variables `response_status` and `response_response`, simply use the response object directly (i.e. `response.status`, `response.response`)